arviz/tests/base_tests/test_data.py

```diff
@@ -323,6 +324,26 @@ def test_addition(self):
         fails = check_multiple_attrs(test_dict, new_idata)
         assert not fails
 
+    @pytest.mark.parametrize("inplace", [True, False])
+    def test_extend_xr_method(self, inplace):
+        data = np.random.normal(size=(4, 500, 8))
+        dataset = from_dict(
+            posterior={"a": data[..., 0], "b": data},
+            prior={"a": data[..., 0], "b": data},
+            posterior_predictive={"a": data[..., 0], "b": data},
+        )
```

Comment:
I think this can be made a fixture, which can then be used with all methods, no need to create the idata every time. see https://github.com/arviz-devs/arviz/pull/1315/files#diff-98d6c3c4d2b318798dcd501599fa5127 with the `data_random`. `test_sel` could also use it so its code can be simplified